### ISSUE TITLE
Fix S2674 FN: Add support for ConfigureAwait

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6517]
### Description

S2674 does not take `ConfigureAwait` into account.

### Repro steps

```cs
  await stream.ReadAsync(array, 0, (int)stream.Length);                       // Raises S2674
  await stream.ReadAsync(array, 0, (int)stream.Length).ConfigureAwait(false); // Does not
```
### Expected behavior

Raise S2674 if ReadAsync is configured to continue on the current synchronization context.

### Actual behavior

Does not raise.

### Known workarounds

No
